peer.go

```diff
@@ -1145,21 +1169,18 @@ func (p *peer) pushAddrMsg(addresses []*btcwire.NetAddress) error {
 			return err
 		}
 		numAdded++
-
-		// Split into multiple messages as needed.
-		if numAdded > 0 && numAdded%btcwire.MaxAddrPerMsg == 0 {
-			p.QueueMessage(msg, nil)
-
-			// NOTE: This needs to be a new address message and not
-			// simply call ClearAddresses since the message is a
-			// pointer and queueing it does not make a copy.
-			msg = btcwire.NewMsgAddr()
-		}
 	}
+	if numAdded > 0 {
+		for _, na := range msg.AddrList {
+			// Add address to known addresses for this peer.
+			p.knownAddresses[addrmgr.NetAddressKey(na)] = struct{}{}
+		}
 
-	// Send message with remaining addresses if needed.
-	if numAdded%btcwire.MaxAddrPerMsg != 0 {
 		p.QueueMessage(msg, nil)
+		// NOTE: This needs to be a new address message and not
```

Comment:
This NOTE is no longer needed since there is only a single message now.  It was there before since multiple messages were being generated.